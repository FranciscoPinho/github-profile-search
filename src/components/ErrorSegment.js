import React from "react";
import { Segment } from "semantic-ui-react";

const ErrorSegment = ({ error }) =>
  error ? (
    <Segment inverted color="red">
      {error}
    </Segment>
  ) : null;

export default ErrorSegment;
