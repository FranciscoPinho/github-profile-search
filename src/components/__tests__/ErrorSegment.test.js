import React from "react";
import { shallow } from "enzyme";
import ErrorSegment from "../ErrorSegment";

const error = `Some error happened`;

describe("ErrorSegment should render correctly", () => {
  it("with error", () => {
    const component = shallow(<ErrorSegment error={error} />);

    expect(component).toMatchSnapshot();
  });
  it("as null with no error", () => {
    const component = shallow(<ErrorSegment />);

    expect(component).toMatchSnapshot();
  });
});
