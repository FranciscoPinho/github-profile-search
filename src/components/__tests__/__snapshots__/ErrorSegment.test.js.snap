// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSegment should render correctly as null with no error 1`] = `""`;

exports[`ErrorSegment should render correctly with error 1`] = `
<Segment
  color="red"
  inverted={true}
>
  Some error happened
</Segment>
`;
